c/shared/source/ufowrite/ufowrite.c

```diff
@@ -340,6 +340,86 @@ int ufwBegFont(ufwCtx h, long flags, char *glyphLayerDir) {
     return ufwSuccess;
 }
 
+static void orderNameKeyedGlyphs(ufwCtx h) {
```

Comment:
I guess we're using these n^2 sorts for stability?

Reply:
I don't think all this sorting is lovely, but cffwrite and t1write have ordering by CID or Name so I half copied what's done in those so that we can correctly sort here. The reason it came up was because we were writing the CIDMap in CID order which then made uforead fail to read it correctly, but more importantly if I open and save the UFO in typical UFO Python tools then the lib.plist was in a completely different. The whole XML dict is sorted by key including CIDMap and we weren't doing that in ufowrite so that's what all this does. Order by name so that we can write the CIDMap correctly and then order by CID so that public.glyphOrder is in the correct CID order. It's kind of ugly. We should really be reading all of this into a libxml2 structure and work from that, but this all works well enough so far.

Reply:
I was thinking more about the efficiency than presence of the sorting, but I guess the worst case is 2^32 operations, which isn't a deal breaker. 

I believe we use a canned quicksort elsewhere, which is why I asked about stability (quicksort isn't stable), but it seems like it's doubtful we'd have duplicate keys anyway (in which case stability isn't relevant). 

I dunno, I guess we can do this for the time being and revisit later.